aws/resource_aws_elasticache_parameter_group.go

```diff
@@ -335,6 +335,10 @@ func resourceAwsElasticacheParameterGroupDelete(d *schema.ResourceData, meta int
 		}
 		return nil
 	})
+	if isResourceTimeoutError(err) {
+		_, err = conn.DeleteCacheParameterGroup(&deleteOpts)
+	}
+	return err
```

Comment:
The above `resource.Retry()` logic is attempting to delete the Elasticache Parameter Group, which may temporarily be in an invalid state due to eventual consistency. In this case, we'll also want to ignore the `CacheParameterGroupNotFoundFault` errors like the `resource.Retry()` does. 👍 

Also as an aside nit, we generally prefer returning context for the error message and using `return nil` at the end of  Terraform resource `Read` and `Delete` functions so it is very obvious that we are exiting in a good state (`err` potentially being `nil` in `return err` is a little ambiguous). (We should probably have a linter checking for this 😉 )

```suggestion
	if isAWSErr(err, elasticache.ErrCodeCacheParameterGroupNotFoundFault, "")
		return nil
	}

	if err != nil {
		return fmt.Errorf("error deleting Elasticache Parameter Group (%s): %s", d.Id(), err)
	}

	return nil
```